src/event/cloudevent.ts

```diff
@@ -46,7 +46,15 @@ export class CloudEvent implements CloudEventV1, CloudEventV03 {
   schemaurl?: string;
   datacontentencoding?: string;
 
-  constructor(event: CloudEventV1 | CloudEventV1Attributes | CloudEventV03 | CloudEventV03Attributes) {
+  /**
+   * Creates a new CloudEvent object with the provided properties. If there is a chance that the event
+   * properties will not conform to the CloudEvent specification, you may pass a boolean `false` as a
+   * second parameter to bypass event validation.
+   *
+   * @param {object} event the event properties
+   * @param {boolean?} strict whether to perform event validation when creating the object - default: true
+   */
+  constructor(event: CloudEventV1 | CloudEventV1Attributes | CloudEventV03 | CloudEventV03Attributes, strict = true) {
```

Comment:
Can we have strict mode default to `false` like is done with web JavaScript?

Reply:
>  like is done with web JavaScript?

What is this referring too?

Reply:
JavaScript is usually opt-in rather than opt-out for things like validation.

I was referring to only being strict when requested like in strict mode:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode

Reply:
not sure that really applies here,  since we aren't talking about the language level strict mode.

Reply:
We're talking about opt-in validation with JS (like JS strict mode).

This PR is opt-out validation, where it could be opt-in validation.